File: lambdas/default/services/base_service.py
# services/base_service.py
from botocore.exceptions import ClientError

class ServiceError(Exception):
    pass


def aws_call(action: callable, error_msg: str):
    try:
        return action()
    except ClientError as e:
        msg = e.response.get("Error", {}).get("Message")
        raise ServiceError(f"{error_msg}: {msg}")

File: apiRest/packages/domain/src/services/baseService.ts
// services/baseService.ts
export class ServiceError extends Error {
    constructor(message: string) {
      super(message);
      this.name = "ServiceError";
    }
  }
  
export async function awsCall<T>(
  action: () => Promise<T>,
  errorMsg: string
): Promise<T> {
  try {
    return await action();
  } catch (err: any) {
    const msg =
      err?.message ||
      err?.name ||
      "Unknown AWS error";

    throw new ServiceError(`${errorMsg}: ${msg}`);
  }
}